core/src/main/java/com/linecorp/armeria/server/encoding/HttpEncodedResponse.java

```diff
@@ -167,9 +167,8 @@ protected void beforeComplete(Subscriber<? super HttpObject> subscriber) {
                 logger.warn("Subscriber.onNext() should not raise an exception. subscriber: {}",
                             subscriber, t);
             }
-        } else {
-            buf.release();
         }
+        buf.release();
```

Comment:
This `buf` must not be released because it's wrapped with an `HttpData` and passed to the subscriber.
The subscriber is responsible for releasing the buf.

I need more information to debug the leak.
- Which Armeria version are you using?
- Do you have any other stack trace?
- Could you show me how you compose the decorators and the service?

Reply:
Hey,

Added the details below:
1. Armeria Version: 1.22.1, but also tested with 1.23.0 (where I faced the same issue)
2. That's the only stacktrace I have, but I have some memory usage charts that point to a leak.
3. Something like this (its spread across multiple files and internal libraries so i'm trying to consolidate and remove internal stuff):
```
val builder: GrpcServiceBuilder = GrpcService.builder()
services.forEach { service: BindableService ->
    builder.addService(ServerInterceptors.intercept(service, interceptors))
}
builder.useBlockingTaskExecutor(true)

val serverBuilder: ServerBuilder = Server.builder()
// add decorators for various things
serverBuilder.service(builder.build())
```

Reply:
Hi, @ashwin1dd, thanks for the information.
But I need more clues for the problem.
Could you try with the latest version of Armeria to see if the leak still happens?
Also, could you provide us with a reproducible example so we can investigate it?

Reply:
Thanks! yeah I'll try that out and get back to you if it still occurs.